gamestate.py

```diff
@@ -144,7 +189,8 @@ class Player(IntEnum):
 
 	max_team_size = 6
 
-	num_player_elements = max_team_size * ATTRIBUTES_PER_POKEMON
+	num_player_elements = (SHARED_ATTRIBUTES_COUNT + TEAM_ATTRIBUTES_COUNT + 
```

Comment:
Doesn't this mean each player will have a separate space for SHARED_ATTRIBUTES when in reality we want just one such space at the top of vector_list. Having this definition of num_player_elements doesn't go hand in hand with _set_weather(), set_player_attributes(), get_player_attributes() etc implementations because there it is assumed that SHARED_ATTRIBUTES are only on the top of the vector_list and not per Pokemon.